tf2_web_republisher/src/tf2_web_republisher.cpp

```diff
@@ -0,0 +1,138 @@
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+
+ *  Author: Julius Kammerl (jkammerl@willowgarage.com)
+ *
+ */
+
+#include "sstream"
+
+#include "tf_web_republisher/tf2_web_republisher.hpp"
+
+#include "geometry_msgs/msg/transform_stamped.hpp"
+#include "rclcpp_action/rclcpp_action.hpp"
+#include "tf2/transform_datatypes.h"
+#include "tf2_ros/buffer.h"
+#include "tf2_web_republisher_msgs/action/tf_subscription.hpp"
+#include "tf2_web_republisher_msgs/srv/republish_t_fs.hpp"
+#include <tf2_ros/transform_listener.h>
+
+TFRepublisher::TFRepublisher(const std::string& name, const rclcpp::NodeOptions& options) : Node(name, options)
+{
+  using namespace std::placeholders;
+
+  action_server_ = rclcpp_action::create_server<tf2_web_republisher_msgs::action::TFSubscription>(
+      this, "tf2_web_republisher",
+      [this](auto&& PH1, auto&& PH2) {
+        return handle_goal(std::forward<decltype(PH1)>(PH1), std::forward<decltype(PH2)>(PH2));
+      },
+      [this](auto&& PH1) { return handle_cancel(std::forward<decltype(PH1)>(PH1)); },
+      [this](auto&& PH1) { handle_accepted(std::forward<decltype(PH1)>(PH1)); });
+
+  tf_buffer_ = std::make_unique<tf2_ros::Buffer>(this->get_clock());
+  tf_listener_ = std::make_shared<tf2_ros::TransformListener>(*tf_buffer_);
+}
+
+rclcpp_action::CancelResponse TFRepublisher::handle_cancel(
+    const std::shared_ptr<rclcpp_action::ServerGoalHandle<tf2_web_republisher_msgs::action::TFSubscription>> /*gh*/)
+{
+  RCLCPP_DEBUG(rclcpp::get_logger("tf2_web_republisher"), "GoalHandle canceled");
+  return rclcpp_action::CancelResponse::ACCEPT;
+}
+
+std::string TFRepublisher::cleanTfFrame(const std::string& frame_id) const
+{
+  if (frame_id[0] == '/')
+  {
+    return frame_id.substr(1);
+  }
+  return frame_id;
+}
+
+std::optional<geometry_msgs::msg::TransformStamped> TFRepublisher::threadSafeLookup(const std::string& target_frame,
+                                                                                    const std::string& source_frame)
+{
+  std::scoped_lock<std::mutex> lock(tf_buffer_mutex_);
+  std::optional<geometry_msgs::msg::TransformStamped> out;
+  try
+  {
+    out = tf_buffer_->lookupTransform(cleanTfFrame(target_frame), cleanTfFrame(source_frame), tf2::TimePointZero);
+  }
+  catch (const tf2::TransformException& ex)
+  {
+    RCLCPP_INFO(this->get_logger(), "Could not transform %s to %s: %s", target_frame.c_str(), source_frame.c_str(),
+                ex.what());
+  }
+
+  return out;
+}
+
+rclcpp_action::GoalResponse
+TFRepublisher::handle_goal(const rclcpp_action::GoalUUID& /*uuid*/,
+                           const std::shared_ptr<const tf2_web_republisher_msgs::action::TFSubscription::Goal>& /*goal*/)
+{
+  RCLCPP_DEBUG(rclcpp::get_logger("tf2_web_republisher"), "GoalHandle request received");
```

Comment:
I don't know loggers very well, but the previous log used just `get_logger()` whereas this one does `rclcpp::get_logger("tf2_web_republisher")` -- chance to just use the previous syntax consistently throughout?

Reply:
+1